src/internal/data/implementations.go

```diff
@@ -51,3 +52,76 @@ func StateOf(prev, curr path.Path) CollectionState {
 
 	return NotMovedState
 }
+
+// -----------------------------------------------------------------------------
+// BaseCollection
+// -----------------------------------------------------------------------------
+
+func NewBaseCollection(
+	curr, prev path.Path,
+	location *path.Builder,
+	ctrlOpts control.Options,
+	doNotMergeItems bool,
+) BaseCollection {
+	return BaseCollection{
+		ctrl:            ctrlOpts,
+		doNotMergeItems: doNotMergeItems,
+		fullPath:        curr,
+		locationPath:    location,
+		prevPath:        prev,
+		state:           StateOf(prev, curr),
+	}
+}
+
+// BaseCollection contains basic functionality like returning path, location,
+// and state information. It can be embedded in other implementations to provide
+// this functionality.
+//
+// Functionality like how items are fetched is left to the embedding struct.
+type BaseCollection struct {
+	ctrl control.Options
```

Comment:
nit: stands out to me now that this var name references the package more than the struct
```suggestion
	opts control.Options
```